invokeai/app/services/image_record_storage.py

```diff
@@ -0,0 +1,353 @@
+
+    def _create_tables(self) -> None:
+        """Creates the tables for the `images` database."""
+
+        # Create the `images` table.
+        self._cursor.execute(
+            f"""--sql
+            CREATE TABLE IF NOT EXISTS images (
+                id TEXT PRIMARY KEY,
+                image_type TEXT, -- non-nullable via foreign key constraint
+                image_category TEXT, -- non-nullable via foreign key constraint
+                session_id TEXT, -- nullable
+                node_id TEXT, -- nullable
+                metadata TEXT, -- nullable
+                created_at TEXT NOT NULL,
+                FOREIGN KEY(image_type) REFERENCES image_types(type_name),
+                FOREIGN KEY(image_category) REFERENCES image_categories(category_name)
+            );
+            """
+        )
+
+        # Create the `images` table indices.
+        self._cursor.execute(
+            """--sql
+            CREATE UNIQUE INDEX IF NOT EXISTS idx_images_id ON images(id);
+            """
+        )
+        self._cursor.execute(
+            """--sql
+            CREATE INDEX IF NOT EXISTS idx_images_image_type ON images(image_type);
+            """
+        )
+        self._cursor.execute(
+            """--sql
+            CREATE INDEX IF NOT EXISTS idx_images_image_category ON images(image_category);
+            """
+        )
+
+        # Create the tables for image-related enums
+        create_enum_table(
+            enum=ImageType,
+            table_name="image_types",
+            primary_key_name="type_name",
+            cursor=self._cursor,
+        )
+
+        create_enum_table(
+            enum=ImageCategory,
+            table_name="image_categories",
+            primary_key_name="category_name",
+            cursor=self._cursor,
+        )
+
+        # Create the `tags` table. TODO: do this elsewhere, shouldn't be in images db service
+        self._cursor.execute(
+            """--sql
+            CREATE TABLE IF NOT EXISTS tags (
+                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                tag_name TEXT UNIQUE NOT NULL
+            );
+            """
+        )
+
+        # Create the `images_tags` junction table.
+        self._cursor.execute(
+            """--sql
+            CREATE TABLE IF NOT EXISTS images_tags (
+                image_id TEXT,
+                tag_id INTEGER,
+                PRIMARY KEY (image_id, tag_id),
+                FOREIGN KEY(image_id) REFERENCES images(id) ON DELETE CASCADE,
+                FOREIGN KEY(tag_id) REFERENCES tags(id) ON DELETE CASCADE
+            );
+            """
+        )
+
+        # Create the `images_favorites` table.
+        self._cursor.execute(
+            """--sql
+            CREATE TABLE IF NOT EXISTS images_favorites (
+                image_id TEXT PRIMARY KEY,
+                favorited_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+                FOREIGN KEY(image_id) REFERENCES images(id) ON DELETE CASCADE
+            );
```

Comment:
It looks like this could just be a field on the `images` table instead of a separate table